include/tvm/runtime/packed_func.h

```diff
@@ -935,6 +952,30 @@ inline TVMType String2TVMType(std::string s) {
     t.bits = 1;
     t.lanes = 1;
     return t;
+  } else if (s.substr(0, 6) == "custom") {
+    // TODO(gus): too much hardcoding here.
+    scan = s.c_str() + 6;
+    if (*scan != '[')
+      LOG(FATAL) << "expected opening brace after 'custom' type in" << s;
+    ++scan;
+    size_t custom_name_len = 0;
+    while (scan + custom_name_len <= s.c_str() + s.length() &&
+           *(scan + custom_name_len) != ']')
+      ++custom_name_len;
+    if (*(scan + custom_name_len) != ']')
+      LOG(FATAL) << "expected closing brace after 'custom' type in" << s;
+    scan += custom_name_len + 1;
+
+    auto type_name = s.substr(7, custom_name_len);
+    // TODO(gus) asserts for these?
+    t.code = GetTypeCode(type_name);
+    t.bits = GetStorageSize(t.code);
```

Comment:
Is this the right place to insert the storage size? It seemed appropriate at the time, and also convenient (because then, later on, when we want to find the underlying storage type size, we can just use `bits()`).